geckomat/change_model/applyKcatConstraints.m

```diff
@@ -80,90 +41,60 @@
-        otherwise % parse 'kcatSource' to extract preferred order
-            sourceOrder = strsplit(kcatSource,'>');
-            kcatSources = model.ec.source(multipleKcatEntries);
-            for j=1:numel(sourceOrder)
-                sourceKcats = strcmp(kcatSources,sourceOrder(j));
-                if any(sourceKcats)
-                    break %If match is found, do not try other sources
-                end
-            end
-    end
-    updateKcats(multipleKcatEntries(~sourceKcats)) = 0;
-    multipleKcatEntries(~sourceKcats)=[]; % Remove sources not to consider
 
-    %For each subunit, first choose the maximum value (for instance from
-    %different substrates)
-    [subunit,firstIdx,subunitIdx] = unique(model.ec.enzyme(multipleKcatEntries),'stable');
-    for j=1:numel(subunit)
-        sameSubunit = multipleKcatEntries(subunitIdx==j);
-        switch kcatSubs
-            case 'max'
-                updateKcats(sameSubunit(1))=max(updateKcats(sameSubunit),[],'omitnan');
-            case 'min'
-                nonZeroKcat = updateKcats(sameSubunit);
-                updateKcats(sameSubunit(1))=min(nonZeroKcat(nonZeroKcat>0),[],'omitnan');
-            case 'median'
-                nonZeroKcat = updateKcats(sameSubunit);                
-                updateKcats(sameSubunit(1))=median(nonZeroKcat(nonZeroKcat>0),'omitnan');
-        end
-        %Set new kcat value at the position of the first occurence of the
-        %subunit, and remove all other kcats for this subunit (for this rxn)
-        sameSubunit(1)=[];
-        updateKcats(sameSubunit)=0; %Remove non-summarized kcat value
+if ~model.ec.geckoLight %For normal GECKO formulation, where each enzyme is explicitly considered
+    %Make vector where column 1 = rxn; 2 = enzyme; 3 = subunit copies; 4 = kcat
+    newKcats=zeros(numel(updateRxns)*10,4);
+    updateRxns=find(updateRxns);
+    kcatFirst=0;
+    for i=1:numel(updateRxns)
+        enzymes   = find(model.ec.rxnEnzMat(i,:));
+        kcatLast  = kcatFirst+numel(enzymes);
+        kcatFirst = kcatFirst+1;
+        newKcats(kcatFirst:kcatLast,1) = updateRxns(i);
+        newKcats(kcatFirst:kcatLast,2) = enzymes;
+        newKcats(kcatFirst:kcatLast,3) = model.ec.rxnEnzMat(i,enzymes);
+        newKcats(kcatFirst:kcatLast,4) = model.ec.kcat(i);
+        kcatFirst = kcatLast;
     end
-    multipleKcatEntries=multipleKcatEntries(firstIdx);
-    switch kcatComplex
-        case 'max'
-            updateKcats(multipleKcatEntries) = max(updateKcats(multipleKcatEntries),[],'omitnan');
-        case 'min'
-            nonZeroKcat = updateKcats(multipleKcatEntries);
-            updateKcats(multipleKcatEntries) = min(nonZeroKcat(nonZeroKcat>0),[],'omitnan');
-        case 'median'
-            nonZeroKcat = updateKcats(multipleKcatEntries);
-            updateKcats(multipleKcatEntries) = median(nonZeroKcat(nonZeroKcat>0),'omitnan');
+    newKcats(kcatLast+1:end,:)=[];
```

Comment:
Perhaps for GeckoLight you could assign "max" as in the code you removed here. I realize it doesn't fit perfectly with the current code, but if you solve it here that way there is no need for special treatment of these reactions (i.e., reactions with GPRs containing OR (that has isozymes))